### PR TITLE
Fixed content type date to save date data-type in phpcr

### PR DESCRIPTION
**Pull request by wachterjohannes**
__tasks:__

- [x] test coverage
- [ ] gather feedback for my changes
- [x] test migration
- [x] add upgrade
- [x] date content type test

__informations:__

| q                | a
| ---------------- | ---
| Fixed tickets    | none
| BC breaks        | see UPGRADE
| Documentation PR | none

### CODE REVIEW

File: src/Sulu/Bundle/ContentBundle/Resources/phpcr-migrations/Version201511171538.php [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+    /**
+     * Migrate the repository up.
+     *
+     * @param SessionInterface $session
+     */
+    public function up(SessionInterface $session)
+    {
+        $this->session = $session;
+        $this->iterateStructures(true);
+        $this->upgradeExternalLinks(true);
```
Comment (danrot) [inline src/Sulu/Bundle/ContentBundle/Resources/phpcr-migrations/Version201511171538.php]: Why have the external links to be upgraded?

File: src/Sulu/Bundle/ContentBundle/Resources/phpcr-migrations/Version201511171538.php [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+                ),
+                'JCR-SQL2'
+            )->execute();
+
+            $name = $this->propertyEncoder->localizedSystemName('external', $localization->getLocalization());
+            foreach ($rows->getNodes() as $node) {
+                /** @var NodeInterface $node */
+                $value = $node->getPropertyValue($name);
+
+                if ($addScheme) {
```
Comment (danrot) [inline src/Sulu/Bundle/ContentBundle/Resources/phpcr-migrations/Version201511171538.php]: Looks like an old migration script to me, which is not related to this PR.

File: src/Sulu/Bundle/ContentBundle/Resources/phpcr-migrations/Version201511171538.php [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+                $value = $this->upgradeProperty($node->getPropertyValue($propertyName), $addScheme);
+                $node->setProperty($propertyName, $value);
+            }
+        }
+    }
+
+    /**
+     * Upgrades the given property to the new date representation.
+     *
+     * @param bool $addScheme Adds the scheme to date if true, removes the scheme otherwise
```
Comment (danrot) [inline src/Sulu/Bundle/ContentBundle/Resources/phpcr-migrations/Version201511171538.php]: What scheme?

Comment (wachterjohannes) [inline src/Sulu/Bundle/ContentBundle/Resources/phpcr-migrations/Version201511171538.php]: this migration script is copied maybe removed to less (:

**Comment by wachterjohannes** [comment]
@danrot fixed

**Pull request merged**